### ISSUE TITLE
Bug in decoding of "scalar" Lepton files

### ISSUE THREAD

**Issue by Melirius** [microsoft/lepton_jpeg_rust#53]
Analyzing merged PR #38 I found out that while current version of library can decode Lepton files, prepared in SIMD version of DropBox Lepton, it cannot decode Lepton files, prepared by its scalar version. 

Steps to reproduce:
- Compile scalar version from https://github.com/dropbox/lepton - it is `lepton-scalar`
- Encode [overflowing file](https://github.com/microsoft/lepton_jpeg_rust/pull/38/files#diff-e9b4242d04be82aa9364648c0adec88d66ce5d91f2080212f8ab58cf70202401) from #38 by `lepton-scalar mathoverflow.jpg mathoverflow_scalar.lep`,  get [mathoverflow_scalar.lep](https://github.com/microsoft/lepton_jpeg_rust/files/13685367/mathoverflow_scalar.tar.gz)

- Try to decode it by current version of `lepton_jpeg_rust` 
``` bash
$ ./lepton_jpeg_util mathoverflow_scalar.lep m.jpg
2023-12-15T12:51:24.441Z INFO  [lepton_jpeg_util::structs::lepton_format] decoding 8 multipexed streams with 8 threads
error code: StreamInconsistent 7 numNonzeros7x7 > 49
```

It appears that `lepton_jpeg_rust`  uses 16 bit SIMD arithmetics in DC estimates prediction ([1](https://github.com/microsoft/lepton_jpeg_rust/blob/cd5160e4ef9e14a4ede81dc435f3de25ec933aed/src/structs/probability_tables.rs#L273), [2](https://github.com/microsoft/lepton_jpeg_rust/blob/cd5160e4ef9e14a4ede81dc435f3de25ec933aed/src/structs/probability_tables.rs#L285)), while "scalar" DropBox Lepton uses 32 bit ([1](https://github.com/dropbox/lepton/blob/f34c7f42d2e0d688239cb880648dc152713c4eae/src/vp8/util/block_context.hh#L47), [2](https://github.com/dropbox/lepton/blob/f34c7f42d2e0d688239cb880648dc152713c4eae/src/vp8/util/block_context.hh#L65)). BTW, as a result [mathoverflow_32.lep](https://github.com/microsoft/lepton_jpeg_rust/pull/38/files#diff-a208518732bd5f83812bfffc206c1adfe6971c155efb0c652bfee6fe4529b997) file encoded by  `lepton_jpeg_rust`  cannot be decoded neither by `lepton-scalar`, nor by `lepton-avx`

**Comment by m6w6** [comment]
Does #54 resp. 413877bfa11d8d8301aac62ab6e9b45a2dfc8a91 fix this issue?

**Comment by Melirius** [comment]
> Does #54 resp. [413877b](https://github.com/microsoft/lepton_jpeg_rust/commit/413877bfa11d8d8301aac62ab6e9b45a2dfc8a91) fix this issue?

No, it is just a library interface change.

**Comment by mcroomp** [comment]
I added #57 to fix this

**Comment by mcroomp** [comment]
fixed with #57
